packages/common/__test__/axios/create-instance.spec.ts

```diff
@@ -89,4 +90,50 @@ describe('src/axios/create-instance', () => {
       expect(onRetry.mock.calls[0][0]).toStrictEqual({rateLimit: 1, delay: 2500})
     })
   })
+
+  describe('Handling response with status != 2xx', () => {
+    it('should throw SellingPartnerApiError instance', async () => {
+      nock('http://www.example.com')
+        .get('/test')
+        .reply(400, 'rate limited')
+
+      const auth = new SellingPartnerApiAuth({refreshToken: ''})
+      const instance = createAxiosInstance({
+        auth,
+        region: 'eu'
+      })
+
+      await expect(instance.get('http://www.example.com/test')).rejects.toThrow(SellingPartnerApiError)
+    })
+
+    it('should have the good message when a response has been sent', async () => {
+      nock('http://www.example.com')
+        .get('/apiName/apiVersion/path')
+        .reply(429, 'rate limited')
+
+      const auth = new SellingPartnerApiAuth({refreshToken: ''})
+      const instance = createAxiosInstance({
+        auth,
+        region: 'eu'
+      })
+
+      await expect(instance.get('http://www.example.com/apiName/apiVersion/path')).rejects.toThrow('apiName (apiVersion) error: Response code 429')
+    })
+
+    it('should have the good message when a response hasn’t been sent', async () => {
+      nock('http://www.example.com')
+        .get('/apiName/apiVersion/pathh')
+        .reply(429, 'rate limited')
+
+      const auth = new SellingPartnerApiAuth({refreshToken: ''})
+      const instance = createAxiosInstance({
+        auth,
+        region: 'eu'
+      })
+
+      await expect(instance.get('http://www.example.com/apiName/apiVersion/path')).rejects.toThrow('apiName error: No response')
```

Comment:
I don’t understand why this one throws `No response`

Reply:
Because `/apiName/apiVersion/path`is different than `/apiName/apiVersion/pathh` (maybe I should change the route name)